### PR TITLE
Closes #2251 Remove block_class config for `az_barrio_search` block and add `az-search-block` class via preprocess hook.

### LINKED ISSUE #2251: Regression: Critical search block class not applied during config distro update using the merge strategy if that block has been edited.
## Problem/Motivation
This merged PR https://github.com/az-digital/az_quickstart/pull/1720 changed how classes are applied to the search form block.
1. Classes are applied to the search form block via block class through config.
2. Classes are removed from the stylesheet.

### Describe the bug
image

Should look like this

image

### To Reproduce

On an existing site (`2.5.x` and older), save this form: `/admin/structure/block/manage/az_barrio_search`
Then apply `2.6.0-rc2` or `2.6.0-rc1` code updates.
Set the config distro strategy to merge
Apply config distro updates.

See broken search form button.
image

## Possible resolutions
1. Revert this commit: https://github.com/az-digital/az_quickstart/commit/ee13d6d398651e4db2498c4c5bd530d87c1b5122
2. Add back the css selectors used before this change themes/custom/az_barrio/css/style.css 
3. Programmatically wrap the block in question with the `az-search-block` class `hook_preprocess_block`
4. Use hook_update to apply precision configuration update adding the class to the block.
5. Add class to the template for the block: https://github.com/az-digital/az_quickstart/blob/main/themes/custom/az_barrio/templates/blocks/block--search-form-block.html.twig

CSS selectors that can be merged image

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Quickstart should not add  classes that the theme relies via block class configuration, unless a reasonable fallback exists.
Block class should be reserved for site builders, and not used for distribution managed css classes, because if a block is edited through the course of normal site building, the merge config strategy will no longer work, and Quickstart can no longer manage those classes.

## Related issues
#2251

## How to test
Patch a site with composer
```
    "extra": {
        "patches": {
            "az-digital/az_quickstart": {
                "Quickstart issue #2251 ": "https://patch-diff.githubusercontent.com/raw/az-digital/az_quickstart/pull/2252.diff"
            }
        }
     }
```

See if it resolves the issue.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
We need to be sure that this fixes the issue when flowing through big pipe as well.
Targeting ID of block, just so we can leave the result exactly the same as the original PR intended.

File: themes/custom/az_barrio/az_barrio.theme [hunk 1]
```diff
@@ -640,4 +640,16 @@ function az_barrio_preprocess_block(&$variables) {
       }
     }
   }
+
+/**
+ * Closes #2251 Remove block_class config foraz_barrio_search block and add
```
Comment (bberndt-uaz) [inline themes/custom/az_barrio/az_barrio.theme:645]:
```suggestion
 * Closes #2251 Remove block_class config for az_barrio_search block and add
```

File: themes/custom/az_barrio/az_barrio.theme [hunk 2]
```diff
@@ -640,4 +640,16 @@ function az_barrio_preprocess_block(&$variables) {
...
+ * az-search-block class via preprocess hook.
+ *
+ * @link https://github.com/az-digital/az_quickstart/pull/2252
```
Comment (bberndt-uaz) [inline themes/custom/az_barrio/az_barrio.theme:648]: Personally, I think we could include just one link. I'd keep the issue link, but either one is definitely fine with me.

```suggestion
```

**Review by bberndt-uaz — APPROVED** [review]
I applied this PR to a multidev that was affected by this issue and confirmed that the `az-search-block` class was added and the block styling was fixed.

**Review by akslay — APPROVED** [review]

**Review by cpoff — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**